services/motion/builtin/move_request.go

```diff
@@ -305,6 +301,47 @@ func (mr *moveRequest) obstaclesIntersectPlan(
 	return state.ExecuteResponse{}, nil
 }
 
+func (mr *moveRequest) augmentBaseExecutionState(
+	baseExecutionState motionplan.ExecutionState,
+) (motionplan.ExecutionState, error) {
+	// update plan
+	existingPlan := baseExecutionState.Plan()
+	newTrajectory := make(motionplan.Trajectory, 0, len(existingPlan.Trajectory()))
+	for idx, currTraj := range existingPlan.Trajectory() {
+		if idx != 0 {
```

Comment:
suppose that we have some plan of the following form:
```
path = [s, p1, p2, g, g]
traj = [  [0,0,0,0], [i1, a1, ds1, de1], 
                     [i2, a2, ds2, de2], 
                     [i3, a3, ds3, de3], [0,0,0,0] ]
```
Then to interpolate correctly across segments we create,
`[0,0,0,0]` is tied to `s`,
`[i1, a1, ds1, de1]` is tied to `s`
`[i2, a2, ds2, de2]` is tied to `p1`
`[i3, a3, ds3, de3]` is tied to `p2`, and
`[0,0,0,0]` is tied to `g`.

Since `s` must appear twice that is why we do `idx--`

Reply:
Ah, this makes sense. We want the inputs of the pose frame to be the start position of that arc, so we must look at the prior step for the starting point. Could we comment to this effect?

[opt] It's fine as-is as this is a private function and it is Known how plans are generated, but consider that this would be slightly more robust if it instead uses `PoseBetweenInverse` on the transform of the traj to the path pose for that index. This would then not break if some future change removed `[0,0,0,0]` as a start set of inputs, or more likely, if some future code uses this function on a subset of a plan.

Reply:
> Could we comment to this effect?
of course, I will give a detailed explanation of this.

> This would then not break if some future change removed [0,0,0,0] as a start set of inputs, or more likely, if some future code uses this function on a subset of a plan.
I agree that with the current impl of this code, what you have written is a limitation. I had this same thought when writing this function but had no clue on how to go about solving this issue. Thank you very much for suggesting this solution. I will update the impl to make it as robust and future-proof as possible :)